### PR TITLE
Remove existing sync lightbulb code

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This helps the work to add lightbulb telemetry, but reducing hte layers of indirections and the need to pass huge numbers of parameters around.

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 1]
```diff
@@ -120,170 +116,13 @@ public bool TryGetTelemetryId(out Guid telemetryId)
...
-
-                    return ConvertToSuggestedActionSets(
-                        state, selection,
-                        fixesTask.WaitAndGetResult(cancellationToken),
-                        refactoringsTask.WaitAndGetResult(cancellationToken),
-                        currentActionCount: 0);
-                }
-            }
-
-            private ImmutableArray<SuggestedActionSet> ConvertToSuggestedActionSets(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:200]: this function was inlined at the callsite.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 2]
```diff
@@ -120,170 +116,13 @@ public bool TryGetTelemetryId(out Guid telemetryId)
...
-                ImmutableArray<UnifiedSuggestedActionSet> fixes,
-                ImmutableArray<UnifiedSuggestedActionSet> refactorings,
-                int currentActionCount)
-            {
-                var filteredSets = UnifiedSuggestedActionsSource.FilterAndOrderActionSets(fixes, refactorings, selection, currentActionCount);
-                return filteredSets.Select(s => ConvertToSuggestedActionSet(s, state.Target.Owner, state.Target.SubjectBuffer)).WhereNotNull().ToImmutableArray();
-            }
-
-            [return: NotNullIfNotNull(nameof(unifiedSuggestedActionSet))]
-            private SuggestedActionSet? ConvertToSuggestedActionSet(UnifiedSuggestedActionSet? unifiedSuggestedActionSet, SuggestedActionsSourceProvider owner, ITextBuffer subjectBuffer)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:212]: this helper moved to be a nested function helper.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 3]
```diff
@@ -120,170 +116,13 @@ public bool TryGetTelemetryId(out Guid telemetryId)
...
-                    {
-                        CodeActionPriority.Lowest => SuggestedActionSetPriority.None,
-                        CodeActionPriority.Low => SuggestedActionSetPriority.Low,
-                        CodeActionPriority.Medium => SuggestedActionSetPriority.Medium,
-                        CodeActionPriority.High => SuggestedActionSetPriority.High,
-                        _ => throw ExceptionUtilities.Unreachable(),
-                    };
-            }
-
-            private static Task<ImmutableArray<UnifiedSuggestedActionSet>> GetCodeFixesAsync(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:263]: moved to be a nested function used directly by it's caller.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 4]
```diff
@@ -300,50 +139,6 @@ private static string GetFixCategory(DiagnosticSeverity severity)
                 }
             }
 
-            private static Task<ImmutableArray<UnifiedSuggestedActionSet>> GetRefactoringsAsync(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:303]: moved to be a nested function used directly by it's caller.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs [hunk 5]
```diff
@@ -158,24 +159,124 @@ await foreach (var set in allSets)
...
+                var refactoringsTask = GetRefactoringsAsync();
 
                 await Task.WhenAll(fixesTask, refactoringsTask).ConfigureAwait(false);
 
                 var fixes = await fixesTask.ConfigureAwait(false);
                 var refactorings = await refactoringsTask.ConfigureAwait(false);
-                foreach (var set in ConvertToSuggestedActionSets(state, selection, fixes, refactorings, currentActionCount))
+
+                var filteredSets = UnifiedSuggestedActionsSource.FilterAndOrderActionSets(fixes, refactorings, selection, currentActionCount);
+                var convertedSets = filteredSets.Select(s => ConvertToSuggestedActionSet(s)).WhereNotNull().ToImmutableArray();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs:179]: this was the helper function inlined.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs [hunk 6]
```diff
@@ -158,24 +159,124 @@ await foreach (var set in allSets)
...
+
+                foreach (var set in convertedSets)
                     yield return set;
+
+                yield break;
+
+                Task<ImmutableArray<UnifiedSuggestedActionSet>> GetCodeFixesAsync()
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs:186]: next four functions were the helpers that were moved.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs [hunk 7]
```diff
@@ -158,24 +159,124 @@ await foreach (var set in allSets)
...
+                    if (owner._codeFixService == null ||
+                        !supportsFeatureService.SupportsCodeFixes(target.SubjectBuffer) ||
+                        !requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.CodeFix))
+                    {
+                        return SpecializedTasks.EmptyImmutableArray<UnifiedSuggestedActionSet>();
+                    }
+
+                    return UnifiedSuggestedActionsSource.GetFilterAndOrderCodeFixesAsync(
+                        workspace, owner._codeFixService, document, range.Span.ToTextSpan(),
+                        priority, options, isBlocking: false, addOperationScope, cancellationToken).AsTask();
```
[the remark this answers is not in the document]
Comment (mavasani) [reply src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs:197]: Actually, ignore my previous comment. Below indicates this flag might still be passed as `true` from TS: https://github.com/dotnet/roslyn/blob/7033bb2ec251386f667c8cf10d745a651ae1f639/src/Workspaces/Core/Portable/CodeFixes/CodeFixContext.cs#L85-L92

[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs:197]: Yup yup.  doing that work in https://github.com/dotnet/roslyn/pull/67549.

**Pull request merged**